packages/data-context/src/util/migrationFormat.ts

```diff
@@ -39,24 +22,26 @@ export const supportFileRegexps: MigrationRegexpGroup = {
     beforeRegexp: 'cypress/\support\/(?<file>index)\.(?=[j|t]s[x]?)',
     afterRegexp: 'cypress/\support\/(?<file>e2e)\.(?=[j|t]s[x]?)',
   },
-}
+} as const
 
-export const regexps: MigrationRegexpGroup = {
+export const regexps = {
   e2e: {
-    beforeRegexp: getLegacyHighlightRegexp('integration'),
-    afterRegexp: getNewHighlightRegexp('e2e'),
-  },
-  component: {
-    beforeRegexp: getLegacyHighlightRegexp('component'),
-    afterRegexp: getNewHighlightRegexp('component'),
+    usingDefaultIntegrationFolder: {
+      beforeRegexp: getLegacyHighlightRegexp('integration'),
+      afterRegexp: getNewHighlightRegexp('e2e'),
+    },
+    usingCustomIntegrationFolder: {
+      beforeRegexp: (folder: string) => `${folder}\/.*?(?<ext>[._-]?[s|S]pec.|[.])(?=[j|t]s[x]?)`,
```

Comment:
can folder contain any special characters? `.` or `/` could probably make it fail right?

Reply:
I have no idea, this regexp has been here for a while. We almost certainly need more unit tests around it. This should be a separate PR, there's likely a ton of complexity to be uncovered when working on this regexp, I will follow it up.

I am thinking the regexp approach may not be ideal.

Reply:
This PR is big, let's do it elsewhere. https://cypress-io.atlassian.net/browse/UNIFY-997